Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -1652,3 +1738,96 @@ fileprivate extension UIViewController {
         dismiss(animated: true)
     }
 }
+
+// MARK: - SettingsViewModel delegation
+extension StatusTableViewController: SettingsViewModelDelegate {
+    func dosingEnabledChanged(_ value: Bool) {
+        DispatchQueue.main.async {
+            self.deviceManager.loopManager.settings.dosingEnabled = value
+        }
+    }
+    
+    func didSave(therapySetting: TherapySetting, therapySettings: TherapySettings) {
+        switch therapySetting {
+        case .glucoseTargetRange:
+            deviceManager?.loopManager.settings.glucoseTargetRangeSchedule = therapySettings.glucoseTargetRangeSchedule
+        case .preMealCorrectionRangeOverride:
+            deviceManager?.loopManager.settings.preMealTargetRange = therapySettings.preMealTargetRange
+        case .workoutCorrectionRangeOverride:
+            deviceManager?.loopManager.settings.legacyWorkoutTargetRange = therapySettings.workoutTargetRange
+        case .suspendThreshold:
+            deviceManager?.loopManager.settings.suspendThreshold = therapySettings.suspendThreshold
+        case .basalRate:
+            deviceManager?.loopManager.basalRateSchedule = therapySettings.basalRateSchedule
+        case .deliveryLimits:
+            deviceManager?.loopManager.settings.maximumBasalRatePerHour = therapySettings.maximumBasalRatePerHour
+            deviceManager?.loopManager.settings.maximumBolus = therapySettings.maximumBolus
+        case .insulinModel:
+            if let insulinModelSettings = therapySettings.insulinModelSettings {
+                deviceManager?.loopManager.insulinModelSettings = insulinModelSettings
+            }
+        case .carbRatio:
+            deviceManager?.loopManager.carbRatioSchedule = therapySettings.carbRatioSchedule
+            deviceManager?.analyticsServicesManager.didChangeCarbRatioSchedule()
+        case .insulinSensitivity:
+            deviceManager?.loopManager.insulinSensitivitySchedule = therapySettings.insulinSensitivitySchedule
+            deviceManager?.analyticsServicesManager.didChangeInsulinSensitivitySchedule()
+        case .none:
+            break // NO-OP
+        }
+    }
```

Comment:
[nit] I wonder why the `StatusTableViewController` needs to handle this logic. Curious if providing the `deviceManager` to the `SettingsViewModel` and letting the view model handle this logic might be a more appropriate approach?

Reply:
@nhamming I hear you... but I'm not a fan of passing around big classes like `DeviceDataManager` everywhere.  That makes it very hard to test.  I may create a protocol that contains `didSave()` and give that to `SettingsViewModel`, and implement this logic in `DeviceDataManager`.  But I consider that a later refactor.